cypress/e2e/tests/migration/migration-waves/crud.test.ts

```diff
@@ -76,8 +78,46 @@ describe(["@tier1"], "Migration Waves CRUD operations", () => {
         cy.get("td", { timeout: 12 * SEC }).should("not.contain", newName);
     });
 
+    // Automates Polarion TC 333
+    it("Migration Wave Application Association", function () {
+        const applications = createMultipleApplications(2);
+        const migrationWave = new MigrationWave(
+            data.getRandomWord(8),
+            now,
+            end,
+            stakeHolders,
+            stakeHolderGroups,
+            applications
+        );
+        migrationWave.create();
+        const applicationTableSelector = `table[aria-label="Applications table for migration wave ${migrationWave.name}"]`;
```

Comment:
we will need to verify this data for other things as well like stakeholders and status.

And all selectors can go into wave.view.ts.

Let's create utility methods in model that can 
wave.verifyApplicationCount(2);
wave.verifyApplicationNames(applicationList)
wave.removeApplications()
wave.delete()

Reply:
Even though the locators are dynamic , they are too big to be put in tests .
wave.selectApplication() or something like that can use this.name to select the application.

Reply:
Makes sense, I'll change it!

Reply:
wave.verifyApplicationCount(2);
wave.verifyApplicationNames(applicationList) 

is for this test only . So yes it can remain here .

wave.applicationsDelete() might be used elsewhere as well though , so we can add a method to delete associated applications.